backend/qemu.pm

```diff
@@ -1127,8 +1127,11 @@ sub handle_qmp_command ($self, $cmd, %optargs) {
         bmwqemu::fctinfo("Skipping the following qmp_command because QEMU_ONLY_EXEC is enabled:\n$line");
         return undef;
     }
-    my $wb = defined $optargs{send_fd} ? tinycv::send_with_fd($sk, $line, $optargs{send_fd}) : syswrite($sk, $line);
-    die "handle_qmp_command: syswrite failed $!" unless ($wb == length($line));
+    local $SIG{__DIE__} = undef;    # prevent custom die handler so that a broken pipe can be handled
+    my $wb = eval { defined $optargs{send_fd} ? tinycv::send_with_fd($sk, $line, $optargs{send_fd}) : syswrite($sk, $line) };
+    return {return => {status => 'shutdown'}} if $@ && $@ =~ qr/Broken pipe/;
```

Comment:
The only problem I see is that encountering a broken pipe does not necessarily mean an expected shutdown happened. So if we call `handle_qml_command` to invoke some random QMP command like we do in many places we will continue the execution as if nothing bad happened in case the pipe broke. This would lead to wrong and maybe not so easy to debug behavior later on.

It would make more sense to me if `power 'off'` would set some flag somewhere and only if this flag is set we would consider broken pipes to be expected and only then we'd ever enter this special error handling branch.

I would also save `$@` and `$!` in variables because it is always a source for error when those are used not directly after the function call that's checked.